### PR TITLE
refactor: Adopt new `addTransaction` option bag changes

### PR DESCRIPTION
**Pull request by OGPoyraz**
**Description**

This PR aims to make mobile client adapt latest version of core `TransactionController` with merged breaking changes. 

Breaking change PR: https://github.com/MetaMask/core/pull/1576
Release PR: https://github.com/MetaMask/core/pull/1630

**Summary:** `addTransaction` is a public API of `TransactionController` consumed by mobile client. 
Old version of function expects second argument `origin` and third `deviceConfirmedOn`. 
New version of the function takes this two argument as an option on second argument. 

All the changes in public API is applied in mobile code. There is no functional change so we expect no change in any flow. 

**Issue**

Fixes https://github.com/MetaMask/mobile-planning/issues/1194

**Checklist**

* [X] There is a related GitHub issue
* [ ] Tests are included if applicable
* [X] Any added code is fully documented

### CODE REVIEW

File: patches/@metamask+transaction-controller+4.0.1.patch [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+new file mode 100644
+index 0000000..5008ddf
+Binary files /dev/null and b/node_modules/@metamask/transaction-controller/.DS_Store differ
 diff --git a/node_modules/@metamask/transaction-controller/dist/EtherscanRemoteTransactionSource.d.ts b/node_modules/@metamask/transaction-controller/dist/EtherscanRemoteTransactionSource.d.ts
 new file mode 100644
 index 0000000..a2d484f
 --- /dev/null
 +++ b/node_modules/@metamask/transaction-controller/dist/EtherscanRemoteTransactionSource.d.ts
-@@ -0,0 +1,15 @@
-+// PATCH GENERATED FROM MetaMask/core branch transaction-controller-patch-v4.0.1-block-tracker
```
Comment (matthewwalsh0) [inline patches/@metamask+transaction-controller+4.0.1.patch:7]: Shall we change the branch name here to our new source of truth so `refactor/transaction-controller-patch-mobile`?

Comment (OGPoyraz) [reply patches/@metamask+transaction-controller+4.0.1.patch:7]: Done

**Comment by OGPoyraz** [comment]
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e6c7c7d1-dfa2-49d9-9442-f0d77feb8756

**Review by matthewwalsh0 — APPROVED** [review]

**Comment by abretonc7s** [comment]
Hi @OGPoyraz @matthewwalsh0 , I see this PR is breaking both sdk and wallet connect which is using the old transaction. I Did it pass linting before merging?

**Pull request merged**